### ISSUE TITLE
HISAT2 2.1.0 fails when options under "Scoring options" used

### ISSUE THREAD

**Issue by jennaj** [galaxyproject/tools-iuc#1792]
Tool version 2.1.0 (most current): https://toolshed.g2.bx.psu.edu/view/iuc/hisat2/40e1083a4430

Test history: https://usegalaxy.org/u/jen/h/test-history-hisat2-soft-clip

 - Dataset 6: uses all defaults and is successful
 - Dataset 7: has the "scoring options" section opened at job submission, but no changes made
 - Dataset 8: has "scoring options" section opened and soft-clip set to Yes (default is No)
 - Datasets 9-18: Test jobs using the two prior versions of the tool (2.0.5.2, 2.0.5.1) are running try to find out when this was introduced. Same test pattern. 
 - Dataset 19: Rerun of dataset 7, but with reference annotation also added under "spliced alignment options" -- expecting it to fail, is just a test to see if a reference annotation (GTF) is for some reason required. If it is, then that input dependency will need to be clarified on the tool form.

I'll also test at https://usegalaxy.eu to make sure this isn't server-specific. The error does not look like a dependency problem, but a tool wrapper problem:

<img width="922" alt="screen shot 2018-03-26 at 11 32 21 am" src="https://user-images.githubusercontent.com/8999735/37925440-671342bc-30e9-11e8-80e8-fe814893d29c.png">

**Comment by jennaj** [comment]
Same results at Galaxy Europe https://usegalaxy.eu/u/jenj/h/test-history-hisat2-soft-clip-eu-server

**Comment by jennaj** [comment]
Reported at https://biostar.usegalaxy.org/p/27285. If I can get around this with an earlier tool version, we can use that as a workaround until 2.1.0 is upgraded/fixed.

**Comment by mvdbeek** [comment]
nah, we can fix it quickly, just running the tests now

**Comment by mvdbeek** [comment]
BTW, all of these `NotFound: cannot find 'xxxx'` are definitely, 100% bugs all of the time -- if you have a means to systematically look at error reports and filter out failed runs for tools (latest version) I think that would be super helpful.

**Comment by jennaj** [comment]
Thank you! Very good to know. I wasn't sure if some errors with this message were server-dependancy problems or not - same error used to point to those (older tool versions, now that I reviewed them). Usage issues previously could also trigger the same message. I'll retest one like that (usage/input problem) in the new tool version using the problematic inputs and see what happens. Sure would simplify troubleshooting if these messages are now restricted to wrapper bugs. 

Btw- All bugs reported from usegalaxy.org, with this error message, using newer tool versions, have tickets already and are confirmed tool bugs. Example: Circos Builder -- I should have put this in IUC repo, but comments already exist about it being a tool problem and it is being addressed, so think I'll leave it there -- unless you think I should move it? https://github.com/galaxyproject/usegalaxy-playbook/issues/105

Going forward, I'll report errors with this message directly to the tool dev repo and skip the vetting to eliminate a server-side issue first. (when bugs are reported is for the newer tool versions)

**Comment by mvdbeek** [comment]
This exact error only happens when a tool parameter is being referenced in the command section that is not declared in the input / configfile or output section. You may need to select a very specific combination of parameters to get there, but if you do it is not a usage problem. Those corner cases sometimes slip through even with our reviews. This error message is not to be confused with `command not found` errors though, that does indeed indicate dependency problems.

**Comment by jennaj** [comment]
Hi @mvdbeek @bgruening - would it be OK to leave this open until the PR is accepted, MTS updated, and then the public servers updated/tested with the fix? 

I pointed some users to this ticket to learn when they can use the tool again with this option.

Alternative - I can create an "check for update and test" for usegalaxy.org under https://github.com/galaxyproject/usegalaxy-playbook/issues and link back here and to my testing summary of items that are not immediately able to be resolved or are in my active to-do testing list https://github.com/galaxyproject/usegalaxy-playbook/issues/86. I plan to convert that into a Project - is getting unwieldly. 

I don't specifically track update testing for usegalaxy.eu but often test there (and in a local) just to compare test results. I haven't been testing in Docker. Is there a repo you keep a track of that sort of thing (issues directly impacting the server)? Or, I could include both server tests in the playbook issues, when changes impact both servers, and test both at the same time.

Or better idea about how to coordinate around this? If there are open, clear, pending tests with existing test histories, this all is much easier to manage and allow others to also test/report back results in a central place. That's what I do in usegalaxy-playbook. I certainly don't want to make final integration testing too complicated - but a queue is nice and I've pretty much always used some version of that on every project I've worked on (whether I was doing the QA myself, or managing those that did) and decided to start using Github to keep track of issues just recently. Works great now, once in a Project will be super easy to manage it al. Note my lists are not comprehensive, the issues are mostly end-user focused (reported problems, or things we found ourselves that once adjusted, will make the Galaxy experience nicer :)
